suplemon/viewer.py

```diff
@@ -621,3 +604,251 @@ def purge_line_cursors(self, line_no):
         for line_cursors in cursor:
             self.remove_cursor(cursor)
         return True
+
+    ###########################################################################
+    # Input Handling
+    ###########################################################################
+
+    def get_key_bindings(self):
+        """Get list of editor key bindings."""
+        return self.app.get_key_bindings()
+
+    def handle_input(self, event):
+        """Handle input."""
+        if event.type == "mouse":
+            return False
+        key = event.key_code
+        name = event.key_name
+        # Try match a key to a method and call it
+
+        key_bindings = self.get_key_bindings()
+        operation = None
+        if key in key_bindings.keys():
```

Comment:
If `key_bindings` is a `dict` (I think it is), then we can use `if key in key_bindings`. The current logic is creating a new list every time.

If this becomes a bottleneck (doubtful), then we can micro-optimize by pre-computing a `set()` which might be more efficient. But once again, prob not a bottleneck